### PR TITLE
[HttpClient] Make RetryableHttpClient implement LoggerAwareInterface

### PR DESCRIPTION
**Pull request by hbgamra**
| Q             | A
| ------------- | ---
| Branch?       | 6.3 for features / 5.4 or 6.2 for bug fixes 
| Bug fix?      | yes/no
| New feature?  | yes/no 
| Deprecations? | yes/no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Can you please explain why you're making this proposal?

File: src/Symfony/Component/HttpClient/RetryableHttpClient.php [hunk 1]
```diff
@@ -28,24 +27,23 @@
...
     /**
      * @param int $maxRetries The maximum number of times to retry
      */
-    public function __construct(HttpClientInterface $client, RetryStrategyInterface $strategy = null, int $maxRetries = 3, LoggerInterface $logger = null)
+    public function __construct(HttpClientInterface $client, RetryStrategyInterface $strategy = null, int $maxRetries = 3)
     {
         $this->client = $client;
         $this->strategy = $strategy ?? new GenericRetryStrategy();
         $this->maxRetries = $maxRetries;
-        $this->logger = $logger ?? new NullLogger();
```
Comment (derrabus) [inline src/Symfony/Component/HttpClient/RetryableHttpClient.php:48]: This change breaks existing code that passes a logger via the constructor and expects it to be used. We cannot do that.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by hbgamra** [comment]
Hello,

I tried to use functions in LoggerAwareTrait without passing through
dependency injection but i think i did a wrong think

 Thank you

Cordially
Housseine

**Comment by nicolas-grekas** [comment]
OK, let me close then.

**Pull request closed without merging**